### DISCUSSION TITLE
Help restoring data from 4.5.10 and 4.6.14

### DISCUSSION THREAD

**Discussion by rfc2119** [osmandapp/OsmAnd#21020] [Q&A] [answered]
Hello all,

I am trying to restore data (favorite locations and recorded tracks) from two OSMAnd+  versions, 4.5.10 and 4.6.14. I had these backed up by [Seedvault](https://github.com/seedvault-app/seedvault). These versions were installed from F-droid. Here's the extracted structure from Seedvault's files:
4.5.10 structure
![4.5.10 structure](https://github.com/user-attachments/assets/5b741117-ae8a-4d44-9402-aac47796eafc)
4.6.14 structure
![4.6.14 structure](https://github.com/user-attachments/assets/d4bfa0b9-8766-4c19-a496-e68fce5e34a9)

The favorite locations and recorded tracks are relatively low. If there's any manual way to restore them I am happy with it. I do not know where to start. The value for `osmand.files:favourites_bak.gpx` (depicted in the screenshots) already contain the favorite locations in XML format. That's a start! but I have no idea how to restore that.

Any guidance/help will be appreciated.

**Comment by MarioSwitch** [comment]
There is a built-in function to Export/Import data: `Hamburger menu > Settings > Import/Export` (at the bottom)

Select Export and then, select all things you want to backups (Favorites, Tracks, etc.), this will create an `.osf` file.
Then import it on the other app and select what you want to import. If there are conflicts, you can skip or replace files.

**Reply by rfc2119** [reply]
Unfortunately I only have the sqlite database (as shown in the screenshots). If there's any way I can download OSMAnd APK and make it read the database file, that'd be great.

**Reply by MarioSwitch** [reply]
Don't you have access to OsmAnd+?

In this case, try replacing content of your files manually, and then reopen app...

**Reply by MarioSwitch** [reply, marked as answer]
> make it read the database file

OsmAnd stores their files in GPX format on your phone/tablet (at `/Android/data/net.osmand/`) and it is possible to edit them manually (generally requires a PC because Android doesn't give you access to this folder).

Based on screenshots, the best way to backup your Favorites is:

1. Close and kill OsmAnd app
2. Connect your phone to your PC and go to  `/Android/data/net.osmand/`
3. Copy Favorites file in your PC (e.g. on the Desktop)
4. Edit the PC copy of the file and replace their contents with "value" of your database (the XML part)
5. Overwrite the phone file by copying the PC file in the same place
6. Open OsmAnd app, you should have your backup favorites shown

**Reply by rfc2119** [reply]
> Don't you have access to OsmAnd+?

I didn't create an account back then. Also this OSMAnd~ version was downloaded from F-droid, not Google Play Store

> Based on screenshots, the best way to backup your Favorites is:

I will try and get back to you. Thanks for sharing

**Reply by rfc2119** [reply]
I couldn't write to `/Android/data/net.osmand.plus/files/` or read from it with a PC. I exported the old "favorite" files to my phone after modifying the XML data to match that of latest version (4.8.6). It turns out OSMAnd can open these files and natively import all the locations. Quite nice!

As for the recorded tracks, I couldn't find them in Seedvault's SQLite database. Maybe I am looking in the wrong path, or maybe OSMAnd does not save recorded trips in `data/net.osmand.plus/files/`. For now I am ok with the results. Thanks for following up @MarioSwitch

**Reply by MarioSwitch** [reply]
> I didn't create an account back then. Also this OSMAnd~ version was downloaded from F-droid, not Google Play Store

You don't need an OsmAnd or Play Store account to export to a file 🤔

---

> As for the recorded tracks, I couldn't find them in Seedvault's SQLite database. Maybe I am looking in the wrong path, or maybe OSMAnd does not save recorded trips in data/net.osmand.plus/files/.

Recorded tracks are saved as files (one file per track) in a sub-folder (named `tracks` iirc)
